interfaces/builtin/serial_port.go

```diff
@@ -208,6 +210,64 @@ func (iface *serialPortInterface) AutoConnect(*snap.PlugInfo, *snap.SlotInfo) bo
 	return true
 }
 
+func (iface *serialPortInterface) HotplugDeviceDetected(di *hotplug.HotplugDeviceInfo) (*hotplug.ProposedSlot, error) {
+	bus, _ := di.Attribute("ID_BUS")
+	if di.Subsystem() != "tty" || bus != "usb" || !serialDeviceNodePattern.MatchString(di.DeviceName()) {
+		return nil, nil
+	}
+
+	slot := hotplug.ProposedSlot{
```

Comment:
For USB devices we could also synthesise a slot label. The value could come from the `ID_MODEL_FROM_DATABASE` and `ID_VENDOR_FROM_DATABASE` attributes. The label is not used for matching but is used for display in some cases (it is also underused because it was always empty).

For example, plugging the Arduino UNO with this change would show a nice human-readable description of the slot.

Reply:
Logic for that already exists and is outside of this PR as a general mechanism for hotplug - see `suggestedSlotName` function in `hotplug.go` - it currently looks at "NAME", "ID_MODEL_FROM_DATABASE", "ID_MODEL" attributes as input data for slot name. That logic doesn't fill in Label and doesn't look at ID_VENDOR_FROM_DATABASE yet, but we can experiment with enhancing it and all interfaces will automatically benefit from it. I think this will be a material for separate PR. The logic here in `HotplugDeviceDetected` shouldn't generally bother with name/label unless an interface wants something non-standard.

Reply:
+1, I would love to see the follow up that uses the label more prominently.

Reply:
Will do, but we will need to think what makes sense here, as we need to populate name and label (let's not discuss here).

Reply:
do I understand correctly that we remember slot names? so if we change naming strategy there will be no confusion/renaming of preexisting slots?

Reply:
@pedronis Yes we remember their names for as long as they have connections, and they wouldn't change if we change naming strategy. The names can only change if you unplug the device and it has no connections, in such case we will forget the slot and it can get a new name next time when device is plugged again.